pkg/cmd/generate-bazel-extra/main.go

```diff
@@ -48,15 +49,54 @@ func runBuildozer(args []string) {
 	}
 }
 
-func getTestTargets(testTargetSize string) ([]string, error) {
-	if _, ok := testSizeToDefaultTimeout[testTargetSize]; !ok {
-		return nil, errors.New("testTargetSize should be one of {small,medium,large,enormous}")
+// parseQueryXML is used because Go doesn't support parsing XML "1.1".
+// It returns a map where the key is a test size {small,medium,large,enormous} and
+// the value is a list of test targets having that size.
+func parseQueryXML(data []byte) (map[string][]string, error) {
+	targetNameRegex, err := regexp.Compile(`<rule class="go_test" location=".*" name="(.*)">`)
+	if err != nil {
+		return nil, err
+	}
+
+	targetSizeRegex, err := regexp.Compile(`<string name="size" value="(.*)"/>`)
+	if err != nil {
+		return nil, err
+	}
+
+	sizeToTargets := map[string][]string{}
+	var currentTargetName string
+	for _, line := range strings.Split(string(data), "\n") {
+		// Check if the line contains a target name.
+		line = strings.TrimSpace(line)
+		submatch := targetNameRegex.FindStringSubmatch(line)
+		if submatch != nil {
+			if currentTargetName != "" {
+				// Default size is medium so if not found then it will be medium.
+				sizeToTargets["medium"] = append(sizeToTargets["medium"], currentTargetName)
+			}
+			currentTargetName = submatch[1]
+			continue
+		}
+		// Check if the line contains a target size.
+		submatch = targetSizeRegex.FindStringSubmatch(line)
+		if submatch != nil {
+			sizeToTargets[submatch[1]] = append(sizeToTargets[submatch[1]], currentTargetName)
+			currentTargetName = ""
+		}
+	}
+	if currentTargetName != "" {
```

Comment:
The repetition here is a little odd but I'm not aware of a great way to avoid it. One thing you can try doing is using a map for target-to-size, then "flipping" the map to size-to-target afterward (this code is not tested):

```
var currentTargetName string
targetToSize := make(map[string]string)
for _, line := range strings.Split(string(data), "\n") {
    ...
    if submatch := targetNameRegex.FindStringSubmatch(line); submatch != nil {
        currentTargetName = submatch[1]
        targetToSize[currentTargetName] = "medium" 
        continue
    }
    if submatch := targetSizeRegex.FindStringSubmatch(line); submatch != nil {
        targetToSize[currentTargetName] = submatch[1]
    }
}
sizeToTarget := make(map[string][]string)
for target, size := range targetToSize {
    sizeToTarget = append(sizeToTarget[size], target)
}
```

Up to you, not going to block the review on it.